filesystem/fs.go

```diff
@@ -167,6 +167,84 @@ func (cfs CFS) list(parentID uint64) ([]fuse.Dirent, error) {
 	return results, nil
 }
 
+// update writes the updated value of 'node'.
+func (cfs CFS) update(node Node) error {
+	inode := node.toJSON()
+	const sql = `
+UPDATE fs.inode SET inode = $1 WHERE id = $2;
+`
+	if _, err := cfs.db.Exec(sql, inode, node.ID); err != nil {
+		return err
+	}
+	return nil
+}
+
+// rename moves 'oldParentID/oldName' to 'newParentID/newName'.
+// If 'newParentID/newName' already exists, it is deleted.
+// See NOTE on node.go:Rename.
+// TODO(marc): this is getting a little gross. We should change this file to
+// be a bunch of methods that take a transaction.
+func (cfs CFS) rename(oldParentID, newParentID uint64, oldName, newName string) error {
+	if oldParentID == newParentID && oldName == newName {
+		return nil
+	}
+
+	tx, err := cfs.db.Begin()
+	if err != nil {
+		return err
+	}
+
+	const lookupSql = `
+SELECT id FROM fs.namespace WHERE (parentID, name) = ($1, $2)`
+
+	// Look for the original ID.
+	var id uint64
+	err = tx.QueryRow(lookupSql, oldParentID, oldName).Scan(&id)
+	if err != nil {
+		_ = tx.Rollback()
+		return err
+	}
+
+	// Delete it.
+	const deleteSql = `
+DELETE FROM fs.namespace WHERE (parentID, name) = ($1, $2);
+`
+	if _, err := tx.Exec(deleteSql, oldParentID, oldName); err != nil {
```

Comment:
Would be better to perform the `DELETE` in the same `Exec` as the `INSERT/UPDATE`. Reduce roundtrips.